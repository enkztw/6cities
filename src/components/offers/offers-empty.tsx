import * as React from 'react';
import {cityName} from '../../types';


const OffersEmtpy: React.FC<{city: cityName}> = ({city}) => {
  return (
    <div className="cities">
      <div className="cities__places-container cities__places-container--empty container">
        <section className="cities__no-places">
          <div className="cities__status-wrapper tabs__content">
            <b className="cities__status">No places to stay available</b>
            <p className="cities__status-description">We could not find any property availbale at the moment in {city}</p>
          </div>
        </section>
        <div className="cities__right-section"></div>
      </div>
    </div>
  );
};

export default React.memo(OffersEmtpy);
