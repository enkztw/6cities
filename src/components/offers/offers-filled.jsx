import React from "react";
import PropTypes from 'prop-types';

import Offer from '../offer/offer.jsx';
import Map from '../map/map.jsx';

class OffersFilled extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      currentOffer: null
    };

    this.handleMouseEnter = this.handleMouseEnter.bind(this);
  }

  handleMouseEnter(e) {
    e.persist();
    e.preventDefault();


    this.setState(() => {
      return {
        currentOffer: e.target
      };
    });
  }

  render() {
    return (
      <div className="cities">
        <div className="cities__places-container container">
          <section className="cities__places places">
            <h2 className="visually-hidden">Places</h2>
            <b className="places__found">{this.props.offers.length} places to stay in {this.props.city}</b>
            <form className="places__sorting" action="#" method="get">
              <span className="places__sorting-caption">Sort by</span>
              <span className="places__sorting-type" tabIndex="0">
                Popular
                <svg className="places__sorting-arrow" width="7" height="4">
                  <use xlinkHref="#icon-arrow-select"></use>
                </svg>
              </span>
              <ul className="places__options places__options--custom">
                <li className="places__option places__option--active" tabIndex="0">Popular</li>
                <li className="places__option" tabIndex="0">Price: low to high</li>
                <li className="places__option" tabIndex="0">Price: high to low</li>
                <li className="places__option" tabIndex="0">Top rated first</li>
              </ul>
            </form>
            <div className="cities__places-list places__list tabs__content">
              {this.props.offers.map((offer) => <Offer {...offer} key={offer.id} />)}
            </div>
          </section>
          <div className="cities__right-section">
            <Map mini={true} offers={this.props.offers}/>
          </div>
        </div>
      </div>
    );
  }
}

OffersFilled.propTypes = {
  city: PropTypes.string,
  offers: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    type: PropTypes.string
  }))
};

export default OffersFilled;
