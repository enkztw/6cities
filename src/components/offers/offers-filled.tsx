import * as React from 'react';

import {Offer} from '../../types';
import withFocusedItem from '../../hocs/with-focused-item/with-focused-item';
import Card from '../offer/offer';
import Map from '../map/map';
import Sorters from '../sorters/sorters';


interface Props {
  offers: Offer[];
  city: string;
  focusedItem: number;
  onBookmarkClick: () => void;
  onMouseEnter: () => void;
  onMouseLeave: () => void;
}

const OffersFilled: React.FC<Props> = ({offers, city, focusedItem, onBookmarkClick, onMouseEnter, onMouseLeave}) => {
  return (
    <div className="cities">
      <div className="cities__places-container container">
        <section className="cities__places places">
          <h2 className="visually-hidden">Places</h2>
          <b className="places__found">
            {offers.length} places to stay in {city}
          </b>
          <Sorters />
          <div className="cities__places-list places__list tabs__content">
            {offers.map((offer) => (
              <Card
                {...offer}
                key={offer.id}
                onBookmarkClick={onBookmarkClick}
                handleMouseEnter={onMouseEnter}
                handleMouseLeave={onMouseLeave}
              />
            ))}
          </div>
        </section>
        <div className="cities__right-section">
          <Map
            mini
            offers={offers}
            focusedOffer={focusedItem}
          />
        </div>
      </div>
    </div>
  );
};

export default withFocusedItem(OffersFilled);
