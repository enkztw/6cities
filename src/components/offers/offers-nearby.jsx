import React from 'react';
import PropTypes from 'prop-types';

import Offer from '../offer/offer.jsx';

function NearbyOffers({offers}) {
  return (
    <div className="container">
      <section className="near-places places">
        <h2 className="near-places__title">Other places in the neighbourhood</h2>
        <div className="near-places__list places__list">
          {offers.map((offer) => <Offer key={offer.id} {...offer} />)}
        </div>
      </section>
    </div>
  );
}

NearbyOffers.propTypes = {
  offers: PropTypes.arrayOf(PropTypes.shape({
    bedrooms: PropTypes.number,
    city: PropTypes.shape({
      location: PropTypes.shape({
        latitude: PropTypes.number,
        longitude: PropTypes.number,
        zoom: PropTypes.number
      }).isRequired,
      name: PropTypes.string
    }),
    description: PropTypes.string,
    goods: PropTypes.arrayOf(PropTypes.string),
    host: PropTypes.shape({
      avatarUrl: PropTypes.string,
      id: PropTypes.number,
      isPro: PropTypes.bool,
      name: PropTypes.string
    }),
    id: PropTypes.number,
    images: PropTypes.arrayOf(PropTypes.string),
    isFavorite: PropTypes.bool,
    isPremium: PropTypes.bool,
    location: PropTypes.shape({
      latitude: PropTypes.number,
      longitude: PropTypes.number,
      zoom: PropTypes.number
    }),
    maxAdults: PropTypes.number,
    previewImage: PropTypes.string,
    price: PropTypes.number,
    rating: PropTypes.number,
    title: PropTypes.string,
    type: PropTypes.string
  }))
};

export default NearbyOffers;
