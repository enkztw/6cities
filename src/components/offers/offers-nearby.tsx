import * as React from 'react';

import {Offer} from '../../types';
import Card from '../offer/offer';


interface Props {
  offers: Offer[];
}

const NearbyOffers: React.FC<Props> = ({offers}: Props) => {
  return (
    <div className="container">
      <section className="near-places places">
        <h2 className="near-places__title">Other places in the neighbourhood</h2>
        <div className="near-places__list places__list">
          {offers.map((offer) => <Card key={offer.id} {...offer} />)}
        </div>
      </section>
    </div>
  );
};

export default React.memo(NearbyOffers);
