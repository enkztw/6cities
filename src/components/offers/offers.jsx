import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import {getOffers} from '../../store/reducers/offers/selectors.js';
import {getCities} from '../../store/reducers/cities/selectors.js';
import {getCity} from '../../store/reducers/cities/selectors.js';

import Tab from '../tab/tab.jsx';
import OffersFilled from './offers-filled.jsx';
import OffersEmpty from './offers-empty.jsx';

function Offers({offers, cities, city}) {
  return (
    <main className={`page__main page__main--index page__main--index-empty${offers.length > 0 ? `` : ` page__main--index-empty`}`}>
      <h1 className="visually-hidden">Cities</h1>
      <div className="tabs">
        <section className="locations container">
          <ul className="locations__list tabs__list">
            {cities.map((it) => <Tab city={it} key={it} />)}
          </ul>
        </section>
      </div>
      {offers.length > 0 ? <OffersFilled offers={offers} city={city} /> : <OffersEmpty city={city} />}
    </main>
  );
}

Offers.propTypes = {
  offers: PropTypes.arrayOf(PropTypes.shape({
    bedrooms: PropTypes.number,
    city: PropTypes.shape({
      location: PropTypes.shape({
        latitude: PropTypes.number,
        longitude: PropTypes.number,
        zoom: PropTypes.number
      }),
      name: PropTypes.string
    }),
    description: PropTypes.string,
    goods: PropTypes.arrayOf(PropTypes.string),
    host: PropTypes.shape({
      avatarUrl: PropTypes.string,
      id: PropTypes.number,
      isPro: PropTypes.bool,
      name: PropTypes.string
    }),
    id: PropTypes.number,
    images: PropTypes.arrayOf(PropTypes.string),
    isFavorite: PropTypes.bool,
    isPremium: PropTypes.bool,
    location: PropTypes.shape({
      latitude: PropTypes.number,
      longitude: PropTypes.number,
      zoom: PropTypes.number
    }),
    maxAdults: PropTypes.number,
    previewImage: PropTypes.string,
    price: PropTypes.number,
    rating: PropTypes.number,
    title: PropTypes.string,
    type: PropTypes.string
  })),
  cities: PropTypes.arrayOf(PropTypes.string),
  city: PropTypes.string
};

const mapStateToProps = (state) => ({
  city: getCity(state),
  cities: getCities(state),
  offers: getOffers(state)
});

export default connect(mapStateToProps)(Offers);


