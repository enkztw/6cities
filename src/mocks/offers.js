export const offers = [
  {
    id: 1,
    name: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    images: [`/img/room.jpg`, `/img/apartment-01.jpg`, `/img/apartment-02.jpg`, `/img/apartment-03.jpg`, `/img/studio-01.jpg`, `/img/apartment-01.jpg`],
    bedrooms: 3,
    adults: 4,
    price: 110,
    premium: false,
    rating: 4.51,
    bookmarked: true
  },
  {
    id: 2,
    name: `Wood and stone place`,
    type: `Room`,
    images: [`/img/apartment-01.jpg`, `/img/room.jpg`, `/img/apartment-02.jpg`, `/img/apartment-03.jpg`, `/img/studio-01.jpg`, `/img/apartment-01.jpg`],
    bedrooms: 1,
    adults: 2,
    price: 120,
    premium: true,
    rating: 4.49,
    bookmarked: true
  },
  {
    id: 3,
    name: `Canal View Prinsengracht`,
    type: `House`,
    images: [`/img/apartment-02.jpg`, `/img/room.jpg`, `/img/apartment-01.jpg`, `/img/apartment-03.jpg`, `/img/studio-01.jpg`, `/img/apartment-01.jpg`],
    bedrooms: 3,
    adults: 3,
    price: 240,
    premium: false,
    rating: 2.11,
    bookmarked: false
  },
  {
    id: 4,
    name: `Nice, cozy, warm big bed apartment`,
    type: `Hotel`,
    images: [`/img/apartment-03.jpg`, `/img/room.jpg`, `/img/apartment-01.jpg`, `/img/apartment-02.jpg`, `/img/studio-01.jpg`, `/img/apartment-01.jpg`],
    bedrooms: 1,
    adults: 2,
    price: 50,
    premium: true,
    rating: 3.2,
    bookmarked: false
  }];
