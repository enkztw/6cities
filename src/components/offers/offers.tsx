import * as React from 'react';
import {connect} from 'react-redux';

import {city} from '../../common/types';
import {Offer} from '../../common/interfaces';
import {getOffers} from '../../store/reducer/offers/selectors';
import {getCities} from '../../store/reducer/cities/selectors';
import {getCity} from '../../store/reducer/cities/selectors';

import Tab from '../tab/tab';
import OffersFilled from './offers-filled';
import OffersEmpty from './offers-empty';


interface Props {
  offers: Offer[];
  cities: string[];
  currentCity: city;
}

const Offers: React.FC<Props> = ({offers, cities, currentCity}) => {
  return (
    <main className={`page__main page__main--index page__main--index-empty${offers.length ? `` : ` page__main--index-empty`}`}>
      <h1 className="visually-hidden">Cities</h1>
      <div className="tabs">
        <section className="locations container">
          <ul className="locations__list tabs__list">
            {cities.map((name) => <Tab name={name} key={name} />)}
          </ul>
        </section>
      </div>
      {offers.length > 0 ? <OffersFilled offers={offers} city={currentCity}/> : <OffersEmpty name={currentCity} />}
    </main>
  );
};

const mapStateToProps = (state) => ({
  currentCity: getCity(state),
  cities: getCities(state),
  offers: getOffers(state)
});

export default connect(mapStateToProps)(Offers);


