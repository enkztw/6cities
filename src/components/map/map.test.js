import React from 'react';
import Enzyme, {mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Map from './map.jsx';

Enzyme.configure({
  adapter: new Adapter()
});

import {offers} from '../../mocks/offers.js';


it(`<Map /> should render`, () => {
  const div = global.document.createElement(`div`);
  global.document.body.appendChild(div);
  const tree = mount(<Map offers={offers} />, {attachTo: div});
  expect(tree.getDOMNode()).toMatchSnapshot();
});
