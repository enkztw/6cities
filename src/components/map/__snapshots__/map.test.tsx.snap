// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Map /> should br rendered 1`] = `
<section
  class="property__map map leaflet-container leaflet-grab leaflet-touch-drag"
  id="map"
  style="position: relative;"
  tabindex="0"
>
  <div
    class="leaflet-pane leaflet-map-pane"
    style="left: 0px; top: 0px;"
  >
    <div
      class="leaflet-pane leaflet-tile-pane"
    >
      <div
        class="leaflet-layer "
        style="z-index: 1;"
      >
        <div
          class="leaflet-tile-container leaflet-zoom-animated"
          style="z-index: 18; left: 0px; top: 0px;"
        >
          <img
            alt=""
            class="leaflet-tile"
            role="presentation"
            src="https://a.basemaps.cartocdn.com/rastertiles/voyager_labels_under/10/525/336{@2x}.png"
            style="width: 256px; height: 256px; left: -237px; top: -134px;"
          />
        </div>
      </div>
    </div>
    <div
      class="leaflet-pane leaflet-shadow-pane"
    />
    <div
      class="leaflet-pane leaflet-overlay-pane"
    />
    <div
      class="leaflet-pane leaflet-marker-pane"
    >
      <img
        alt=""
        class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
        src="img/pin.svg"
        style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -162px; top: 18px; z-index: 18;"
        tabindex="0"
      />
      <img
        alt=""
        class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
        src="img/pin.svg"
        style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -31px; top: -24px; z-index: -24;"
        tabindex="0"
      />
      <img
        alt=""
        class="leaflet-marker-icon leaflet-zoom-hide leaflet-interactive"
        src="img/pin.svg"
        style="margin-left: -13.5px; margin-top: -19.5px; width: 27px; height: 39px; left: -31px; top: -24px; z-index: -24;"
        tabindex="0"
      />
    </div>
    <div
      class="leaflet-pane leaflet-tooltip-pane"
    />
    <div
      class="leaflet-pane leaflet-popup-pane"
    />
  </div>
  <div
    class="leaflet-control-container"
  >
    <div
      class="leaflet-top leaflet-left"
    />
    <div
      class="leaflet-top leaflet-right"
    />
    <div
      class="leaflet-bottom leaflet-left"
    />
    <div
      class="leaflet-bottom leaflet-right"
    >
      <div
        class="leaflet-control-attribution leaflet-control"
      >
        <a
          href="https://leafletjs.com"
          title="A JS library for interactive maps"
        >
          Leaflet
        </a>
      </div>
    </div>
  </div>
</section>
`;
